src/Modules/CalcOffence.lua

```diff
@@ -3815,37 +3815,31 @@ function calcs.offence(env, actor, activeSkill)
 			durationMod = m_max(durationMod, 0)
 			local rateMod = calcLib.mod(skillModList, cfg, "BleedFaster") + enemyDB:Sum("INC", nil, "SelfBleedFaster")  / 100
 			globalOutput.BleedDuration = durationBase * durationMod / rateMod * debuffDurationMult
-			local bleedStacks = (output.HitChance / 100) * (globalOutput.BleedDuration / (output.HitTime or output.Time) * skillData.dpsMultiplier) / maxStacks
+
+			-- The average number of bleeds that will be active on the enemy at once
+			local bleedStacks = (output.HitChance / 100) * (output.BleedChanceOnHit / 100) * (globalOutput.BleedDuration / (output.HitTime or output.Time) * skillData.dpsMultiplier)
```

Comment:
output.HitChance is not the right multiplier if you have perfect agony. it should probably be replaced by some mix of output.HitChance*output.BleedChanceOnHit and output.CritChance*output.BleedChanceOnCrit.

Reply:
Fixed perfect agony and similar effects.